source/app/service-providers/provider-contract.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * @ignore
+ * BEGIN HEADER
+ *
+ * Contains:        ProviderContract
+ * CVM-Role:        Class Contract
+ * Maintainer:      Hendrik Erz
+ * License:         GNU GPL v3
+ *
+ * Description:     This class is the base for all service providers and defines
+ *                  their external API.
+ *
+ * END HEADER
+ */
+
+export default abstract class ProviderContract {
+  /**
+   * Service providers implement a boot method which must be called after
+   * instantiation of the singleton.
+   *
+   * @return  {Promise<void>}  boot must return a Promise.
+   */
+  public abstract boot (): Promise<void>
```

Comment:
Maybe provide a default implementation for the boot and shutdown methods, since some of the providers don't need them and then you don't need to implement them.

Reply:
Good idea